popup.js

```diff
@@ -0,0 +1,159 @@
+document.addEventListener('DOMContentLoaded', function () {
+  var appsBtn = document.getElementById('apps');
+  appsBtn.addEventListener('click', function () {
+    chrome.runtime.sendMessage({
+      type: 'set_current_category',
+      category: 'apps'
+    });
+  });
+
+  var codingBtn = document.getElementById('coding');
+  codingBtn.addEventListener('click', function () {
+    chrome.runtime.sendMessage({
+      type: 'set_current_category',
+      category: 'coding'
+    });
+  });
+
+  var travelBtn = document.getElementById('travel');
+  travelBtn.addEventListener('click', function () {
+    chrome.runtime.sendMessage({
+      type: 'set_current_category',
+      category: 'travel'
+    });
+  });
+
+  var recipesBtn = document.getElementById('recipes');
+  recipesBtn.addEventListener('click', function () {
+    chrome.runtime.sendMessage({
+      type: 'set_current_category',
+      category: 'recipes'
+    });
+  });
+
+  var moviesBtn = document.getElementById('movies');
+  moviesBtn.addEventListener('click', function () {
+    chrome.runtime.sendMessage({
+      type: 'set_current_category',
+      category: 'movies'
+    });
+  });
+
+  var booksBtn = document.getElementById('books');
+  booksBtn.addEventListener('click', function () {
+    chrome.runtime.sendMessage({
+      type: 'set_current_category',
+      category: 'books'
+    });
+  });
+  
+  var podcastsBtn = document.getElementById('podcasts');
+  podcastsBtn.addEventListener('click', function () {
+    chrome.runtime.sendMessage({
+      type: 'set_current_category',
+      category: 'podcasts'
+    });
+  });
+
+  var sportsBtn = document.getElementById('sports');
+  sportsBtn.addEventListener('click', function () {
+    chrome.runtime.sendMessage({
+      type: 'set_current_category',
+      category: 'sports'
+    });
+  });
+
+  var musicBtn = document.getElementById('music');
```

Comment:
There's a lot of repetition, you can modularize this configuration (creating the buttons) with a function:

```js
function createButton(id) {
  var button = document.getElementById(category);
  button.addEventListener('click', function (category) {
    chrome.runtime.sendMessage({
      type: 'set_current_category',
      category: category
    });
  });
}
```